public/app/features/logs/components/log-context/LogRowContextModal.tsx

```diff
@@ -317,10 +315,22 @@ export const LogRowContextModal: React.FunctionComponent<LogRowContextModalProps
 
     const currentGen = generationRef.current;
     try {
-      const newRows = await loadMore(place);
+      // we consider all the currently existing rows, even the original row,
+      // this way this array of rows will never be empty
+      const allRows = [...above.rows, row, ...below.rows];
+
+      const newRows = await loadMore(place, allRows);
+      const [older, newer] = partition(newRows, (newRow) => newRow.timeEpochNs > row.timeEpochNs);
+      const newAbove = logsSortOrder === LogsSortOrder.Ascending ? newer : older;
+      const newBelow = logsSortOrder === LogsSortOrder.Ascending ? older : newer;
       if (currentGen === generationRef.current) {
-        setSection(place, (section) => ({
-          rows: place === 'above' ? [...newRows, ...section.rows] : [...section.rows, ...newRows],
+        setSection('above', () => ({
```

Comment:
i recommend just doing a `setContext` here that sets both places, to guarantee both change at the same atomic time, so something like:
```ts
setContext({
    above : ...,
   below: ....,
});
```

Reply:
Good point, updated in 9ba878fae6105a71635adfabb71b9dbeb80be36a